### PR TITLE
New HDFS Spout

### PR DESCRIPTION
**Pull request by calzadaharo**
This is the first version of an HDFS Spout. It can ingest both individual or partitioned files.

### CODE REVIEW

File: src/main/scala/com/raphtory/spouts/HdfsSpout.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    def nextLine(): String = {
+      bufferReader.readLine()
+    }
+
+    def finished(line: String): Boolean = {
+      if (line == null) {
+        if(status.length > 0) {
+          hdfsManager = HDFSManager()
+          println("Still loading Spout...")
```
Comment (miratepuffin) [inline src/main/scala/com/raphtory/spouts/HdfsSpout.scala:44]: Hi Rodrigo, thanks for adding this in! Would you mind changing these prints to use loggers and I shall merge it in :)

Comment (calzadaharo) [reply src/main/scala/com/raphtory/spouts/HdfsSpout.scala]: Done! I hope it is what you were hoping for.

Comment (miratepuffin) [reply src/main/scala/com/raphtory/spouts/HdfsSpout.scala]: Fab! Will merge it in now 🎉

**Review by miratepuffin — APPROVED** [review]

File: src/main/scala/com/raphtory/spouts/HdfsSpout.scala (hunk 1 above)
Comment (calzadaharo) [reply src/main/scala/com/raphtory/spouts/HdfsSpout.scala]: Nice! :)

**Pull request merged**